src/wui/interactive_player.cc

```diff
@@ -571,14 +571,22 @@ void InteractivePlayer::draw_map_view(MapView* given_map_view, RenderTarget* dst
 
 			const Image* player_image = nullptr;
 			float icon_scale = 0.7f;
+			float icon_opacity = 1.0f;
 
 			// Not all map starting positions pass the suitability test.
 			// TODO(tothxa): Make the editor at least use the same test. But manual changes would still
 			//               be possible.
 			for (unsigned p = map.get_nrplayers(); p != 0u; --p) {
 				if (map.get_starting_pos(p) == f->fcoords) {
-					player_image = playercolor_image(p - 1, icon_filename);
+					Widelands::Player* plr = gbase.get_player(p);
+					if (plr == nullptr || !plr->is_picking_custom_starting_position()) {
+						// Should have a HQ if finished picking, no need for the overlay
```

Comment:
Does this mean that "plr" has a hq anywhere and we do not show his colour in his default start position?

I do not know for what reason the default starting position is shown in the player colour. We can do this check or skip it, no meaning about.

Reply:
I think it may help to know the starting positions recommended by the map author.

Reply:
> > Does this mean that "p" has a hq anywhere and we do not show his colour in his default start position?
> 
> The default starting position is replaced by the picked one, so we could only show it over the HQ, but that does not make much sense IMO.
> 

I did not know that the default starting position will be replaced by the picked one. (Where in the code ...) But a try confirmed it. In this case it makes sense.

> > I do not know for what reason the default starting position is shown in the player colour. We can do this check or skip it, no meaning about.
> 
> I think it may help to know the starting positions recommended by the map author.

Agree.